cpp-lite/include/mxnet-lite/symbol_block.h

```diff
@@ -0,0 +1,151 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*!
+*  Copyright (c) 2016 by Contributors
+* \file symbol.h
+* \brief definition of symbol
+* \author Chuntao Hong, Zhang Chen
+*/
+
+#ifndef MXNET_LITE_SYMBOL_BLOCK_H_
+#define MXNET_LITE_SYMBOL_BLOCK_H_
+
+#include <map>
+#include <unordered_map>
+#include <string>
+#include <vector>
+
+#include <mxnet-lite/base.h>
+#include <mxnet-lite/ndarray.h>
+#include <mxnet-lite/symbol.h>
+
+namespace mxnet {
+namespace lite {
+
+class SymbolBlock {
+ public:
+  SymbolBlock(const Context& context,
```

Comment:
As we discussed, we would need an interface without distinguishing `data` and `params`, something like this:
```cpp
SymbolBlock(const Context& context,
            const std::vector<std::pair<std::string, std::string> >& flags,
            const Symbol& output,
            std::vector<NDArray>& inputs);  // the order of inputs follows output.ListInputNames(kAll)
```

Reply:
you can pass {} to params